File: packages/ios-sdk/Sources/OpencomSDK/Core/OpencomError.swift
import Foundation

/// Errors that can occur when using the Opencom SDK.
public enum OpencomError: LocalizedError {
    case notInitialized
    case noVisitor
    case networkError(Error)
    case invalidResponse
    case pushNotificationsFailed(Error)
    case invalidWorkspace

    public var errorDescription: String? {
        switch self {
        case .notInitialized:
            return "Opencom SDK is not initialized. Call Opencom.initialize() first."
        case .noVisitor:
            return "No visitor session available."
        case .networkError(let error):
            return "Network error: \(error.localizedDescription)"
        case .invalidResponse:
            return "Invalid response from server."
        case .pushNotificationsFailed(let error):
            return "Push notification registration failed: \(error.localizedDescription)"
        case .invalidWorkspace:
            return "Invalid workspace ID."
        }
    }
}

File: packages/android-sdk/sdk/src/main/kotlin/com/opencom/sdk/OpencomError.kt
package com.opencom.sdk

/**
 * Errors that can occur when using the Opencom SDK.
 */
sealed class OpencomError : Exception() {

    /**
     * SDK has not been initialized. Call Opencom.initialize() first.
     */
    data object NotInitialized : OpencomError() {
        private fun readResolve(): Any = NotInitialized
        override val message: String = "SDK not initialized. Call Opencom.initialize() first."
    }

    /**
     * No visitor session is available.
     */
    data object NoVisitor : OpencomError() {
        private fun readResolve(): Any = NoVisitor
        override val message: String = "No visitor session available."
    }

    /**
     * Invalid workspace ID provided.
     */
    data object InvalidWorkspace : OpencomError() {
        private fun readResolve(): Any = InvalidWorkspace
        override val message: String = "Invalid workspace ID."
    }

    /**
     * Network request failed.
     */
    data class NetworkError(override val cause: Throwable?) : OpencomError() {
        override val message: String = "Network request failed: ${cause?.message}"
    }

    /**
     * API returned an error response.
     */
    data class ApiError(val statusCode: Int, val errorMessage: String?) : OpencomError() {
        override val message: String = "API error ($statusCode): ${errorMessage ?: "Unknown error"}"
    }

    /**
     * Failed to parse API response.
     */
    data class ParseError(override val cause: Throwable?) : OpencomError() {
        override val message: String = "Failed to parse response: ${cause?.message}"
    }
}
